db/active_replicator_common.go

```diff
@@ -473,3 +581,10 @@ func removeLocalStatus(ctx context.Context, metadataStore base.DataStore, status
 	_, _, err = putDocWithRevision(metadataStore, statusKey, currentStatus.Rev, nil, 0)
 	return err
 }
+
+// registerFunctions must be called once after immediately after newActiveReplicatorCommon to set the functions that require a circular definition from parent (ActiveReplicatorPush/ActiveReplicatorPull) and activeReplicatorCommon.
+func (apr *activeReplicatorCommon) registerFunctions(getStatusFn func() *ReplicationStatus, connectFn func() error, registerCheckpointerCallbacksFn func(*activeReplicatorCollection) error) {
+	apr._getStatusCallback = getStatusFn
+	apr.replicatorConnectFn = connectFn
+	apr.registerCheckpointerCallbacksFn = registerCheckpointerCallbacksFn
```

Comment:
```suggestion
func (a *activeReplicatorCommon) registerFunctions(getStatusFn func() *ReplicationStatus, connectFn func() error, registerCheckpointerCallbacksFn func(*activeReplicatorCollection) error) {
	a._getStatusCallback = getStatusFn
	a.replicatorConnectFn = connectFn
	a.registerCheckpointerCallbacksFn = registerCheckpointerCallbacksFn
```

Reply:
Decided to name `arc` as method name because that is a bit more clear and matches `active_replicator_collections.go`